### PR TITLE
Refactor typography

### PR DESCRIPTION
**Pull request by 36degrees**
This refactors the way that typography works within Frontend to take into account Dave's work on styling prose.

As part of this work we have also:

- Adopted responsive typography 'maps' which define the way type should behave at different breakpoints in a single place, rather than multiple declarations of sizes and line heights we had in Frontend Toolkit
- Made font smoothing an integral part of the 'core' typography rather than relying on individual parts of the core or components applying the mixin
- Replaced usage of the word 'core' with 'regular' throughout
- Added new overrides for font size / line height and font weight.
- Removed 'font-size-adjust' (for now) – see commit ec4f497 for details.

## Still to do

- [x] Default line heights (and font sizes) for print media need to be reviewed by @dashouse

### CODE REVIEW

File: src/globals/scss/core/_typography.scss [hunk 1]
```diff
@@ -1,435 +1,165 @@
...
-  }
-  font-weight: $font-weight;
-  text-transform: none;
+  // Headings
 
-  @if $is-print {
-    font-size: $font-size-print;
-    line-height: $line-height;
+  .govuk-heading-xl {
+    @include govuk-font-bold-48;
```
Comment (dashouse) [inline src/globals/scss/core/_typography.scss]: I believe all of the headings, paragraph classes that use HTML elements need to be normalised. At the moment `<h1 class="govuk-heading-xl">govuk-heading-xl</h1>` will have the default margin-top applied to the `h1`

Comment (36degrees) [reply src/globals/scss/core/_typography.scss]: Have pushed a further commit to remove top margins from all headings and paragraphs

File: src/globals/scss/core/_typography.scss [hunk 2]
```diff
@@ -1,435 +1,173 @@
...
     }
-
   }
 
-  .heading-small {
-    @include govuk-bold-19;
-
-    margin-top: em($govuk-spacing-scale-2, 16);
-    margin-bottom: em($govuk-spacing-scale-1, 16);
+  .govuk-body-m  + .govuk-heading-l,
```
Comment (dashouse) [inline src/globals/scss/core/_typography.scss]: We're missing the contextual adjustments for `govuk-body`, the alt class for `govuk-body-m`

Comment (36degrees) [reply src/globals/scss/core/_typography.scss]: Good spot!

Comment (36degrees) [reply src/globals/scss/core/_typography.scss]: Have pushed a commit to fix that.

File: src/globals/scss/core/_typography.scss [hunk 3]
```diff
@@ -1,435 +1,173 @@
...
-      margin-top: em($govuk-spacing-scale-4, 19);
+      padding-top: $govuk-spacing-scale-4;
     }
   }
 
-  .body-text {
-    @include govuk-core-19;
-    margin-top: em($govuk-spacing-scale-1, 16);
-    margin-bottom: em($govuk-spacing-scale-4, 16);
+  .govuk-body-m + .govuk-heading-m,
```
Comment (dashouse) [inline src/globals/scss/core/_typography.scss]: Apologies I missed this... this should also include `govuk-body-m` + `govuk-heading-s` and `govuk-body-s` + `govuk-heading-s`

Comment (36degrees) [reply src/globals/scss/core/_typography.scss]: Pushed a fix

**Comment by dashouse** [comment]
Been experimenting with this and not had any problems that haven't already been addressed in the additional commits.

Example usage:

gov uk frontend typography

**Comment by 36degrees** [comment]
@dashouse What do you want to do about the print styles? Specifically with regards to the line height which I've currently just defaulted to 1.

**Comment by 36degrees** [comment]
@hannalaakso I've rebased this on master to resolve some conflicts - the last 3 commits (
d224717, 50d469e, a4f5da1) are new since you last reviewed – could you check you're happy with them?

Then I think this is good to merge!

**Pull request merged**